codex/raw/marketplace.md

```diff
@@ -0,0 +1,269 @@
+
+## Motivation
+The Codex network aims to create a peer-to-peer storage engine with strong data durability, 
+data persistence guarantees and node storage incentives.
+To reach this goal, a data availability and retrieval mechanism is needed.
+Support for light clients, like mobile devices should also be embraced.
+The protocol should remove complexity to allow for simple implementation and 
+simplify incentive mechanisms.
+
+## Semantics 
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+| Terminology  | Description |
+| --------------- | --------- |
+| storage providers | A Codex node that provides storage services to the marketplace. |
+| validator nodes | A Codex node that checks for missing storage proofs and triggers for a reward. |
+| client nodes | The most common Codex node that interacts with other nodes to store, locate and retrieve data. |
+| slots | Created by client nodes when a new dataset is requested to be stored. Discussed further in the [slots section](#slots).  |
+
+### Storage Request
+
+Client nodes can create storage requests on the Codex network via the Codex marketplace.
+The marketplace is a smart contract deployed on a EVM-compatiable blockchain.
+The smart contract MUST handle storage request, storage slot state, 
+storage provider rewards, storage provider collateral, and storage proof state. 
+
+To create a request to store a dataset on the Codex network,
+client nodes MUST split the dataset into data chunks, $(c_1, c_2, c_3, \ldots, c_{n})$.
+Using an erasure coding technique, 
+the data chunks are encoded and placed into separate slots.
+The erasure coding technique SHOULD be the [Reed-Soloman algorithm](https://hackmd.io/FB58eZQoTNm-dnhu0Y1XnA).
+
+The requester, client node, SHOULD submit a transaction with the desired request parameters.
+The requester MUST provide a `duration` value for the storage request along with the appropriate `reward`,
+payment for request. 
+Once a request is created via the transaction, 
+all slots MUST be filled by storage providers before the request is officially started.
+If the request does not attract enough storage providers after a time set by the requester,
+which is defined by as `expiry`,
+the request MUST be `canceled`.
+If canceled, 
+`collateral` SHOULD be returned to all storage providers that participated in the initiation of the request, 
+including a portion of the `reward`. 
+The remaining `reward` is returned to the requester.
+The requester MAY create a new request with different values to restart the process.
+
+The storage request SHOULD initiate a new storage contract with the following values:
+
+```solidity
+
+  // the Codex node requesting storage
+  address client;
+
+  // content identifier
+  string cid;
```

Comment:
I've noticed this before, but forgot to bring it up - we should not store the Cid as a string but as a byte array/string. It should be changed in the contract implementation and should be correctly captured here.

Reply:
Good point, we should be using [the binary format](https://github.com/multiformats/cid#how-does-it-work) instead of the stringified form. Adam already created an issue for this: https://github.com/codex-storage/codex-contracts-eth/issues/107